File: Adapter/Adapter/EnemyRobotAdapter.cs
using Adapter.Product;

namespace Adapter.Adapter {
    internal class EnemyRobotAdapter : IEnemyAttacker {
        EnemyRobot robot;

        public EnemyRobotAdapter(EnemyRobot newRobot) {
            this.robot = newRobot;
        }

        public void ShootBullet() {
            robot.SlashWithSword();
        }

        public void DriveForward() {
            robot.WalkForward();
        }

        public void AssignDriver(string driverName) {
            robot.ReactToHuman(driverName);
        }
    }
}

File: Adapter.java/src/com/company/EnemyRobotAdapter.java
package com.company;

public class EnemyRobotAdapter implements  EnemyAttacker{
    EnemyRobot robot;

    public EnemyRobotAdapter(EnemyRobot newRobot) {
        this.robot = newRobot;
    }

    @Override
    public void shootBullet() {
        robot.slashWithSword();
    }

    @Override
    public void driveForward() {
        robot.walkForward();
    }

    @Override
    public void assignDriver(String driverName) {
        robot.reactToHuman(driverName);
    }
}
